src/discordcr/rest.cr

```diff
@@ -577,20 +579,20 @@ module Discord
     # Server" permission.
     #
     # [API docs for this method](https://discordapp.com/developers/docs/resources/guild#modify-guild)
-    def modify_guild(guild_id : UInt64, name : String?, region : String?,
-                     verification_level : UInt8?, afk_channel_id : UInt64?,
-                     afk_timeout : Int32?, icon : String?, owner_id : UInt64?,
-                     splash : String?)
-      json = {
-        name:               name,
-        region:             region,
+    def modify_guild(guild_id : UInt64, name : String? = nil, region : String? = nil,
+                     verification_level : UInt8? = nil, afk_channel_id : UInt64? = nil,
+                     afk_timeout : Int32? = nil, icon : String? = nil, owner_id : UInt64? = nil,
```

Comment:
So, something I realized, is that if you actually want to *remove* a guilds icon (reset it to the default), in API terms you send:
```json
{
  "icon": null
}
```
which would mean passing `icon: nil` - but this would not get emitted by the JSON builder.

To actually remove the icon, this is also valid:
```json
{
  "icon": ""
}
```
Which we can do, and won't get wiped out by the builder via `icon: ""` in the method call.

While this is still fine (discarding nulls > strict arguments) it might be worth documenting this in case someone actually wants to do this, as it's not obvious IMO. Maybe with a `NOTE:` docstring.

This also applies to a user's `avatar`.